pkg/buildpipeline/pipeline.go

```diff
@@ -535,24 +535,37 @@ type TransformedStage struct {
 	// TODO: Most blocks for Sequential/Parallel handling are identical. Maybe
 	// just use a single `Nested []TransformedStage` field and have a `Type string`
 	// field or similar to distinguish between sequential/parallel.
-	Sequential []TransformedStage
-	Parallel   []TransformedStage
+	Sequential []*TransformedStage
+	Parallel   []*TransformedStage
 	Task       *pipelinev1alpha1.Task
 	// PipelineTask is non-empty only if Task is non-empty, but it is populated
 	// after Task is populated so the reverse is not true.
 	PipelineTask *pipelinev1alpha1.PipelineTask
 	// The depth of this stage in the full tree of stages
 	Depth int8
+	// The stage enclosing this stage, or nil if it is at top level
+	EnclosingStage *TransformedStage
+	// The stage immediately before this stage at the same depth, or nil if there is no such stage
+	PreviousSiblingStage *TransformedStage
+	// TODO: Add the equivalent reverse relationship
+}
+
+func (ts TransformedStage) isSequentialStage() bool {
```

Comment:
I'd just shorten these names to `isSequential` and `isParallel` - the `Stage` part is kinda implicit. =)

Reply:
Yeah I was thinking about doing the same, will do that now.